ironmon_tracker/screens/UpdateScreen.lua

```diff
@@ -0,0 +1,309 @@
+
+	local startX = Constants.SCREEN.WIDTH + Constants.SCREEN.MARGIN + 15
+	local startY = Constants.SCREEN.MARGIN + 65
+	for _, button in ipairs(UpdateScreen.OrderedMenuList) do
+		button.box = { startX, startY, 110, 16 }
+		startY = startY + 21
+	end
+
+	for _, button in pairs(UpdateScreen.Buttons) do
+		button.type = Constants.ButtonTypes.FULL_BORDER
+		button.textColor = "Lower box text"
+		button.boxColors = { "Lower box border", "Lower box background" }
+	end
+
+	-- These buttons share a location, but visiable at different times based on Update Status
+	UpdateScreen.Buttons.ReloadTracker.box = UpdateScreen.Buttons.RemindMeLater.box
+	UpdateScreen.Buttons.ManualDownload.box = UpdateScreen.Buttons.RemindMeLater.box
+
+	if Main.OS ~= "Windows" then
+		UpdateScreen.Buttons.UpdateNow.text = "Open download link"
+	end
+end
+
+function UpdateScreen.performAutoUpdate()
+	UpdateScreen.currentState = UpdateScreen.States.IN_PROGRESS
+	Program.redraw(true)
+
+	-- Disable Bizhawk sound while the update is in process
+	local wasSoundOn = client.GetSoundOn()
+	client.SetSoundOn(false)
+
+	gui.clearImageCache() -- Unsure if this is needed
+	emu.frameadvance() -- Required to allow the redraw to occur before batch commands begin
+
+	-- Execute the batch set of operations
+	local success = UpdateScreen.executeBatchOperations()
+	UpdateScreen.currentState = Utils.inlineIf(success, UpdateScreen.States.SUCCESS, UpdateScreen.States.ERROR)
+	Program.redraw(true)
+
+	if client.GetSoundOn() ~= wasSoundOn then
+		client.SetSoundOn(wasSoundOn)
+	end
+
+	if UpdateScreen.currentState == UpdateScreen.States.SUCCESS then
+		Main.Version.showUpdate = false
+		Main.SaveSettings(true)
+	end
+end
+
+function UpdateScreen.executeBatchOperations()
+	-- For non-Windows OS, likely need to use something other than a .bat file
+	if Main.OS ~= "Windows" then
+		return false
+	end
+
+	-- Temp Files/Folders used by batch operations
+	local archiveName = "Ironmon-Tracker-main.tar.gz"
+	local folderName = "Ironmon-Tracker-main"
+
+	-- Each individual command listed in order, to be appended together later
+	local batchCommands = {
+		'echo Downloading the latest Ironmon Tracker version.',
+		string.format('curl -L "%s" -o "%s"', Constants.Release.TAR_URL, archiveName),
+		'echo Extracting downloaded files.',
+		string.format('tar -xf "%s" && del "%s"', archiveName, archiveName),
+		'echo Applying the update; copying over files.',
+		string.format('rmdir "%s\\.vscode" /s /q', folderName),
+		string.format('rmdir "%s\\ironmon_tracker\\Debug" /s /q', folderName),
+		string.format('del "%s\\.editorconfig" /q', folderName),
+		string.format('del "%s\\.gitattributes" /q', folderName),
+		string.format('del "%s\\.gitignore" /q', folderName),
+		string.format('del "%s\\README.md" /q', folderName),
+		string.format('xcopy "%s" /s /y /q', folderName),
+		string.format('rmdir "%s" /s /q', folderName),
+		'echo Version update completed successfully.',
+		'timeout /t 3',
+	}
```

Comment:
I know you mentioned it'd be nice for better error info if something went wrong, here's a suggestion:

Pause the commands when an error occurs. When you're chaining commands like this, if you want to run a command if any of the previous commands failed you can do so like this:
```cmd
(command 1 && command 2 && command 3 && ...) || <command to run if error>
```
So in the lua script could implement it as:
```lua
local batchCommands = {
    '(echo Downloading the latest Ironmon Tracker version.',
    ... --(the other commands)
    'timeout /t 3) || pause',
}
```
Note that the opening bracket goes before the first command and the closing bracket after the last command, since they get joined later. The `||` specifies "run the following command only if the previous command failed", which in this instance the "previous command" is actually the entire batch of commands. This pauses the command output leaving the window open with the error displayed for someone to screenshot and post when asking for help on the issue (instead of it just instantly closing without any indication of what went wrong). 

For example, here I changed the `TAR_URL` to download a tar from a different branch than main, meaning `archiveName` and `folderName` were not correct:
> ![image](https://user-images.githubusercontent.com/106463662/200885575-a6e5cc3a-4712-459a-ac38-a4aca29c6369.png)

Weirdly though it seems to error there when attempting to copy the files, not on extracting? lol 

If you want to instead just output the errors to a log similarly to the randomiser error log the syntax is very similar:
```cmd
(command 1 && command 2 && command 3 && ...) 2>outputfile.txt
```
though what i noticed this way is that the table for the curl output download gets sent to the output file instead of the command prompt, but that doesn't really matter for the average user

Reply:
Good notes here, this is something I definitely want to include. I went with option 1 for now, just having it pause if there was an error.